staging/src/k8s.io/client-go/tools/portforward/portforward.go

```diff
@@ -349,6 +359,10 @@ func (pf *PortForwarder) handleConnection(conn net.Conn, port ForwardedPort) {
 	errorStream, err := pf.streamConn.CreateStream(headers)
 	if err != nil {
 		runtime.HandleError(fmt.Errorf("error creating error stream for port %d -> %d: %v", port.Local, port.Remote, err))
+		_ = pf.streamConn.Close()
+		if pf.streamFailChan != nil {
+			close(pf.streamFailChan)
+		}
```

Comment:
this is the more concerning change as is hard to predict all the situations that will trigger this code path , what can make CreateStream to fail?  are there errors that are retryable here?

Reply:
The code changes proposed in this PR are as follows:
- stop closing `streamConn` on every error, that's the change you highlighted below 
- now, we need to have a way to identify when a connection is broken, that's where this piece of code into play, it will trigger every time whenever we have a problem establishing a new stream, which will happen when a remote end is broken (pod removal, network error, etc), or when the local end of the connection is interrupted (network error). 

> are there errors that are retryable here?

Based on my testing I haven't identify any that could be retryable, but that doesn't mean there aren't any. So once we identify we can always expand the code here. For now, I believe this is a good starting point.